### ISSUE TITLE
Correct path for kpn_snappy header/footer

### ISSUE THREAD

**Issue by TELLO0815** [Sylius/InvoicingPlugin#119]
SyliusInvoicePlugin / kpn Snappy can not get the header from twig. I createt a testheader in same folder like pdf.html.twig. /src/Resources/SyliusInvoicingPlugin/views/Invoice/Download 

But its not working to render them in my pdf.

```
knp_snappy:
    pdf:
        enabled:    true
        binary:     '%env(WKHTMLTOPDF_PATH)%'
        options:
            - { name: 'header-html', value: '@SyliusInvoicingPlugin/Invoice/Download/_header.html.twig' }
```
  
But if i write html direkt in the knp_snappy.yaml it works.  

```
- { name: 'header-html', value: '<!DOCTYPE html>
                                             <html>
                                                 <body>
                                                     <div style="border: 5px dashed crimson; color: maroon; background-color: darksalmon">
                                                         This is a header
                                                     </div>
                                                 </body>
                                             </html>' }
```

Any idea?

I found a solution, but not so nice.

no twig and I must put the path like this:

```  - { name: 'header-html', value: `'%kernel.project_dir%/src/Resources/SyliusInvoicingPlugin/views/Invoice/Download/_header.html' } ```

**Comment by bartoszpietrzak1994** [comment]
Hello TELLO0815!

Which version of InvoicingPlugin are you currently using?

**Comment by TELLO0815** [comment]
v0.8.3

**Comment by bartoszpietrzak1994** [comment]
Hello TELLO0815!

I've investigated your problem and passing the template path with `@SyliusInvoicingPlugin` prefix doesn't work for me either.

Unfortunately, I have not found any other Symfony and knp_snappy users with the same problem as yours but I assume that the missing part here is calling the Twig templating engine with the header path as an argument.

Therefore, I changed the logic in the `InvoicePdfFileGenerator`'s `generate()` method a bit:

```
$pdf = $this->pdfGenerator->getOutputFromHtml(
            $this->templatingEngine->render($this->template, [
                'invoice' => $invoice,
                'channel' => $channel,
                'invoiceLogoPath' => $this->fileLocator->locate($this->invoiceLogoPath),
            ]), 
            ['header-html' => $this->templatingEngine->render('@SyliusInvoicingPlugin/Invoice/Download/_header.html.twig')]
        );
```

and it works like a charm. For now I think that there are two ways of generating the invoice's header via knp_snappy bundle:

1) Pass the path to the file using `%kernel.project.dir%` parameter, just like you did in your solution
2) Overwrite the default `InvoicePdfFileGenerator` class implementation with a customized one using the snipped provided above.
